tests/device_tests/test_msg_stellar_sign_transaction.py

```diff
@@ -318,3 +318,22 @@ def test_manage_data(client):
         == b"MTa8fmhM7BxYYo9kY1RKQFFrg/MVfzsgiPSuP7+SJZjhQ3P1ucN5JUnfmD5484ULRPjVTd+0YYjS6hScLupbCQ=="
         # 3136bc7e684cec1c58628f6463544a40516b83f3157f3b2088f4ae3fbf922598e14373f5b9c3792549df983e78f3850b44f8d54ddfb46188d2ea149c2eea5b09
     )
+
+
+def test_sign_tx_manage_buy_offer_op(client):
+    op = messages.StellarManageBuyOfferOp()
+    op.buying_asset = messages.StellarAssetType(0)
+    op.amount = 200
+    op.selling_asset = messages.StellarAssetType(
+        1, "X", "GAUYJFQCYIHFQNS7CI6BFWD2DSSFKDIQZUQ3BLQODDKE4PSW7VVBKENC"
+    )
+    op.buying_price = 10
+
+    tx = _create_msg()
+
+    response = stellar.sign_tx(client, tx, [op], ADDRESS_N, NETWORK_PASSPHRASE)
+
+    assert (
+        b64encode(response.signature)
+        == b"JBd7YKhZYqRho0nvO6w+r0XD8RiJsNQXrQgWjmo8NimxaEblpOloAHTTA1JMATh0kbhBxbyRnJ1yLAfdzrSBAw=="
```

Comment:
test on legacy is failing here: https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/384534467
part of it is the operation type, but I fixed that locally and the signature is still different. Leaving this for you to investigate because I don't know what should be correct.

Reply:
Possibly that `offer_id = 0` thing.

Reply:
no, the test is still failing with your current changes.
it seems that `price_n` and `price_d` are missing.
AFAICT the C code does not check for presence of fields and uses 0s everywhere, but Python will crash on it.

Reply:
correct